src/relay/op/dyn/tensor/transform.cc

```diff
@@ -467,6 +467,9 @@ bool StridedSliceRel(const Array<Type>& types, int num_inputs, const Attrs& attr
   int64_t num_axis = dshape.size();
 
   const auto* begin = types[1].as<TensorTypeNode>();
+  if (begin == nullptr) {
+    return false;
+  }
```

Comment:
This and the change below in `src/relay/op/tensor/transform.cc` are the fix for the type inference issue mentioned in "Known issues" section of https://github.com/apache/tvm/pull/9746

No test is added because it is hard to reproduce on a simple test case and the change is trivial.